jazz_codeq/test/test.js

```diff
@@ -0,0 +1,757 @@
+			fromDateStub = sinon.stub(validation, "validateFromDate").returns('fromdate');
+			toDateStub = sinon.stub(validation, "validateToDate").returns('todate');
+			fromAfterToStub = sinon.stub(validation, "validateFromAfterTo").returns(true);
+		});
+
+		afterEach(() => {
+			sandbox.restore();
+			event = {};
+			config = {};
+			query = {};
+			getQueryStub.restore();
+			missingFieldsStub.restore();
+			fromDateStub.restore();
+			toDateStub.restore();
+			fromAfterToStub.restore();
+			getMetricsStub.restore();
+		});
+
+		it('should return error when getMetrics is called for pathString "codeq"', () => {
+			getMetricsStub = sinon.stub(utils, "getMetrics").returns({error: "Error getting metrics"});
+			const result = index.getCodeqInputsUsingQuery(event, config);
+
+			sinon.assert.calledOnce(getQueryStub);
+			sinon.assert.calledOnce(missingFieldsStub);
+			sinon.assert.calledOnce(fromDateStub);
+			sinon.assert.calledOnce(toDateStub);
+			sinon.assert.calledOnce(fromAfterToStub);
+			sinon.assert.calledOnce(getMetricsStub);
+			
+			expect(result.error).to.eq("Error getting metrics");
+		});
+
+		it('should return data when getCodeq report is called for pathString "codeq"', () => {
+			getMetricsStub = sinon.stub(utils, "getMetrics").returns({metrics:['code-coverage']});
+			const result = index.getCodeqInputsUsingQuery(event, config)
+			
+			sinon.assert.calledOnce(getQueryStub);
+			sinon.assert.calledOnce(missingFieldsStub);
+			sinon.assert.calledOnce(fromDateStub);
+			sinon.assert.calledOnce(toDateStub);
+			sinon.assert.calledOnce(fromAfterToStub);
+			sinon.assert.calledOnce(getMetricsStub);
+			
+			expect(result.metrics[0]).to.eq('code-coverage');
+		});
+	});
+	
+	describe('getReportOnError tests', () => {
+		let metrics = [];
+		let serviceContext = {};
+
+		beforeEach(() => {
+			metrics = ['code-coverage'];
+			serviceContext = { query: 'codeq'};
+		});
+
+		afterEach(() => {
+			metrics = [];
+			serviceContext = {};
+		})
+
+		it('should return report data when error code is 404 for getReportOnError', () => {
+			const err = {
+				report_error: 'report error',
+				code: 404
+			};
+			
+			const getReportStub = sinon.stub(utils, 'getReport').resolves({output: 'report data' });
+
+			index.getReportOnError(err, metrics, {}, serviceContext)
+			.then(result => {
+				sinon.assert.calledOnce(getReportStub);
+
+				expect(result.data.output).to.eq("report data");
+
+				getReportStub.restore();
+			});
+		});
+
+		it('should return report error when getReport rejects for getReportOnError', () => {
+			const err = {
+				report_error: 'report error',
+				code: 404
+			};
+			
+			const getReportStub = sinon.stub(utils, 'getReport').rejects({report_error: 'report error' });
+
+			index.getReportOnError(err, metrics, {}, serviceContext)
+			.then(result => {
+				sinon.assert.calledOnce(getReportStub);
+
+				expect(result.error).to.eq("report error");
+
+				getReportStub.restore();
+			});
+		});
+

```

Comment:
How about a test case when it is an error and response code is not `404`?

Reply:
added